src/speechly/client.ts

```diff
@@ -211,27 +217,43 @@ export class Client {
   }
 
   /**
-   * Stops current SLU context by sending a stop context event to the API and muting the microphone.
+   * Stops current SLU context by sending a stop context event to the API and muting the microphone
+   * delayed by contextStopDelay = 250 ms
    */
   async stopContext(): Promise<string> {
     if (this.state !== ClientState.Recording) {
       throw Error('Cannot stop context - client is not recording')
     }
 
     this.setState(ClientState.Stopping)
-    this.microphone.mute()
+    const contextId: string = await new Promise((resolve) => {
+      Promise.race([
+        new Promise((resolve) => setTimeout(resolve, this.contextStopDelay)), // timeout
+        new Promise((resolve) => { this.resolveStopContext = resolve }),
```

Comment:
We should make sure that we call `clearTimeout` here in case this path gets called.

Reply:
Yes, i will add one more promise here. Since the timeout Promise only in use in the race, we can fire and forget it.

Reply:
Actually, no need in another one promise, I will just store the promise of the contextId and check it (and wait for it) in startContext.